alibi/explainers/similarity/grad.py

```diff
@@ -5,82 +5,139 @@
-        sim_fn:
-            Similarity function to use. ``'grad_dot'`` | ``'grad_cos'``. Default: ``'grad_dot'``.
-        task:
-            Task to perform. ``'classification'`` | ``'regression'``. Default: ``'classification'``.
-        store_grads:
-            Whether to store gradients. Default ``False``. If ``False``, gradients are computed on the fly otherwise we
-            store them which can be faster when it comes to explaining and instances.
-        seed:
-            Random seed. Default: 0.
-        backend:
-            Backend to use. ``'tensorflow'`` | ``'torch'``. Default: ``'tensorflow'``.
-        kwargs:
-            Additional arguments to pass to the similarity function.
+        predictor
+            Model to explain.
+        loss_fn
+            Loss function used. The gradient of the loss function is used to compute the similarity between the test
+            instances and the training set. This should be the same loss used to train the model.
+        sim_fn
+            Similarity function to use.
+        task
+            Task performed by the model.
+        store_grads
+            Whether to store gradients. If ``False``, gradients are computed on the fly otherwise we store them which
+            can be faster when it comes to computing explanations. Storing gradients may be memory intensive if the
+            model is large.
+        backend
+            Backend to use.
+        device
+            Device to use. If ``None``, the default device for the backend is used. If using `pytorch` backend see
+            `pytorch device docs <https://pytorch.org/docs/stable/tensor_attributes.html#torch-device>`_ for correct
+            options. Note that in the `pytorch` backend case this parameter can be a ``torch.device``. If using
+            `tensorflow` backend see `tensorflow docs <https://www.tensorflow.org/api_docs/python/tf/device>`_ for
+            correct options.
         """
-
-        self.meta = copy.deepcopy(DEFAULT_META_SIM)
-        self.meta['params'].update(
-            sim_fn_name=sim_fn,
-            store_grads=store_grads,
-            seed=seed,
-            backend_name=backend,
-            task_name=task
-        )
+        # TODO: add link to docs page for GradientSimilarity explainer in the docstring
 
         sim_fn_opts: Dict[str, Callable] = {
             'grad_dot': dot,
             'grad_cos': cos,
             'grad_asym_dot': asym_dot
         }
 
-        if isinstance(sim_fn, str):
-            if sim_fn not in sim_fn_opts.keys():
-                raise ValueError(f'Unknown method {sim_fn}. Consider using: `{"` | `".join(sim_fn_opts.keys())}`.')
+        if sim_fn not in sim_fn_opts.keys():
+            raise ValueError(f"""Unknown method {sim_fn}. Consider using: '{"' | '".join(sim_fn_opts.keys())}'.""")
+
+        resolved_sim_fn = sim_fn_opts[sim_fn]
 
-            sim_fn = sim_fn_opts[sim_fn]
+        if task not in Task.values():
```

Comment:
Considering my other comment, if `Task` inherits from `str` then can do `if task not in Task.__members__.values()`